src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_PROPERTY_VALUE = 100;
+    private static final int NUMBER_OF_FIGURES = 5;
+
+    public Figure getRandomFigure() {
+        String color = new ColorSupplier().getRandomColor();
+        Random random = new Random();

```

Comment:
move this to field level
this way you will not create redundant objects each time `getRandomFigure()` is called